sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/http/client/DefaultHttpClientProvider.java

```diff
@@ -1,20 +1,23 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-package com.generic.core.http.client;
+package com.generic.core.implementation.http.client;
 
+import com.generic.core.http.client.DefaultHttpClientBuilder;
+import com.generic.core.http.client.HttpClient;
+import com.generic.core.http.client.HttpClientProvider;
 import com.generic.core.util.configuration.Configuration;
 
 /**
  * An {@link HttpClientProvider} that provides an implementation of HttpClient based on native JDK HttpURLConnection.
  */
-class DefaultHttpClientProvider implements HttpClientProvider {
+public class DefaultHttpClientProvider implements HttpClientProvider {
     private static final boolean ENABLE_HTTP_CLIENT_SHARING =
         Configuration.getGlobalConfiguration().get("ENABLE_HTTP_CLIENT_SHARING", Boolean.FALSE);
     private final boolean enableHttpClientSharing;
 
     // Enum Singleton Pattern
-    private enum GlobalHttpUrlConnectionHttpClient {
+    public enum GlobalHttpUrlConnectionHttpClient {
```

Comment:
```suggestion
    public enum GlobalDefaultHttpClient {
```

Reply:
Why does this need to be public? Can we not have a `private static final GLOBAL_HTTP_CLIENT` that is returned from `getInstance()`?